src/librustc_mir/borrow_check/nll/region_infer/error_reporting/region_name.rs

```diff
@@ -558,9 +593,18 @@ impl<'tcx> RegionInferenceContext<'tcx> {
             return None;
         }
 
+        let mir_node_id = tcx.hir.as_local_node_id(mir_def_id).expect("non-local mir");
+        let args_span = if let hir::ExprKind::Closure(_, _, _, span, _)
+            = tcx.hir.expect_expr(mir_node_id).node
+        {
+            span
+        } else {
+            mir.span
```

Comment:
We can do better here, too. `mir.span` is pretty horrific in all cases. Maybe we leave that for another issue though. We would want to extract (from the HIR) the span of the return type, I suppose. Really, we should do the same "return type is `&'1 u32`" sort of thing we do for arguments, right?

I wonder if, for closures, we can highlight just the final `|`?

So it would look like:

```
error: unsatisfied lifetime constraints
  --> $DIR/E0621-does-not-trigger-for-closures.rs:25:26
   |
LL |     invoke(&x, |a, b| if a > b { a } else { b }); //~ ERROR E0495
   |                    --    ^^^^^ requires that `'1` must outlive `'2`
   |                    ||
   |                    |return type of closure is `&'2 i32`
   |                    has type `&'1 i32`
```

Reply:
This is the code to extract the type name:

https://github.com/rust-lang/rust/blob/4b8089daf8046d7999310d44e5c68ccff4ab255a/src/librustc_mir/borrow_check/nll/region_infer/error_reporting/region_name.rs#L278-L280

Reply:
To get the final character you can use [the `end_point` method](https://doc.rust-lang.org/nightly/nightly-rustc/syntax/codemap/struct.CodeMap.html#method.end_point)